src/H5Dbtree2.c

```diff
@@ -854,7 +920,10 @@ H5D__bt2_idx_insert(const H5D_chk_idx_info_t *idx_info, H5D_chunk_ud_t *udata,
     assert(H5_addr_defined(udata->chunk_block.offset));
 
     /* Check if the v2 B-tree is open yet */
-    if (NULL == idx_info->storage->u.btree2.bt2) {
+    if (H5D__bt2_idx_is_open(idx_info, &index_open) < 0)
```

Comment:
I'm a little nervous about potential function call overhead here (and other places that call this function within this file). It might get inlined, but maybe we should create a macro here to be safe?

Reply:
I'd be very surprised if there was any overhead at all, but I could create a macro that the is_open function calls as well.